ha/core/controllers/pcs/cluster_controller.py

```diff
@@ -119,19 +134,41 @@ def storageset_list(self) -> dict:
         raise HAUnimplemented("This operation is not implemented.")
 
     @controller_error_handler
-    def add_node(self, nodeid: str = None, descfile: str = None) -> dict:
+    def add_node(self, node_id: str = None, cluster_user: str = None,
+                 cluster_password: str = None, desc_file: str = None) -> dict:
         """
         Add new node to cluster.
-
+        :param cluster_user:
+        :param cluster_password:
+        :param node_id:
+        :param desc_file:
         Args:
-            nodeid (str, optional): Provide nodeid. Defaults to None.
-            filename (str, optional): Provide descfile. Defaults to None.
+            node_id (str, optional): Provide node_id. Defaults to None.
+            filename (str, optional): Provide desc_file. Defaults to None.
 
         Returns:
             ([dict]): Return dictionary. {"status": "", "msg":""}
                 status: Succeeded, Failed, InProgress
         """
-        raise HAUnimplemented("This operation is not implemented.")
+        if not node_id and not desc_file:
+            return {"status": "Failed", "msg": "Either node_id or desc_file is required to add node"}
+
+        if desc_file:
+            _json_data = PcsClusterController.load_json_file(desc_file)
+            node_id = _json_data.get("node_id")
+            cluster_user = _json_data.get("cluster_user")
+            cluster_password = _json_data.get("cluster_password")
+        elif node_id and not(cluster_user or cluster_password):
+            return {"status": "Failed", "msg": "Missing parameters (cluster_user or cluster_password) for node_id"}
+
+        self.auth_node(node_id, cluster_user, cluster_password)
```

Comment:
No error checking here. What if auth_node fails?

Reply:
We are checking error in auth_node itself and throwing ClusterManagerError which at the end is caught by our generic annotation @controller_error_handler

Reply:
we are getting msg like below in response: